lib/BootBot.js

```diff
@@ -14,20 +14,45 @@ class BootBot extends EventEmitter {
     if (!options || (options && (!options.accessToken || !options.verifyToken || !options.appSecret))) {
       throw new Error('You need to specify an accessToken, verifyToken and appSecret');
     }
+    this._options = options
     this.accessToken = options.accessToken;
     this.verifyToken = options.verifyToken;
     this.appSecret = options.appSecret;
     this.broadcastEchoes = options.broadcastEchoes || false;
     this.app = express();
     this.webhook = options.webhook || '/webhook';
     this.webhook = this.webhook.charAt(0) !== '/' ? `/${this.webhook}` : this.webhook;
-    this.app.use(bodyParser.json({ verify: this._verifyRequestSignature.bind(this) }));
     this._hearMap = [];
     this._conversations = [];
   }
-
+  setBeforeMiddlewares( beforeMiddlewares ){
+    if( !( beforeMiddlewares instanceof Array ) ){
+      throw new Error( 'Paremeter in setBeforeMiddlewares() must be an array of middlewares' )
+    }else if( !beforeMiddlewares || !beforeMiddlewares.length ){
+      return
+    }
+    beforeMiddlewares.forEach( middleware => {
+      this.app.use( middleware )
+    })
+  }
+  setAfterMiddlewares( afterMiddlewares ){
+    if( !( afterMiddlewares instanceof Array ) ){
+      throw new Error( 'Paremeter in setAfterMiddlewares() must be an array of middlewares' )
+    }else if( !afterMiddlewares || !afterMiddlewares.length ){
+      return
+    }
+    afterMiddlewares.forEach( middleware => {
+      this.app.use( middleware )
+    })
+  }
```

Comment:
These two methods are identical, you can just have a `setMiddlewares()` method and you just call it before and after initializing the webhooks with different params.